### PR TITLE
update some deps and stuff

### PR DESCRIPTION
**Pull request by oponder**
Updates various dependencies and fixes some errors / warnings in the tests.

Also removes some unused dependencies.

### CODE REVIEW

**Comment by oponder** [comment]
Hmm ok storybook doesn't like one of the changed web pack deps. will check it out tmw.

File: package.json [hunk 1]
```diff
@@ -85,10 +85,8 @@
     "@airbrake/browser": "2.1.3",
     "@hot-loader/react-dom": "17.0.1",
     "axios": "0.21.1",
-    "bluebird": "3.7.2",
```
Comment (axbarsan) [inline package.json:88]: Wasn't this required as a peer dependency of `giantswarm-js-client`?

Comment (oponder) [reply package.json:88]: I don't think so... 🕵️ it was introduced here: https://github.com/giantswarm/happa/pull/578/files which was about the original app catalog, but I probably didn't end up using it? Not sure...but as far as I can tell Happa works fine without it.

**Comment by marians** [comment]
ICYM, pushing to catalogs currently fails. https://gigantic.slack.com/archives/C3C7ZQXC1/p1617811021034800

**Review by marians — APPROVED** [review]

**Pull request merged**